tls/extensions/s2n_extension_list.c

```diff
@@ -176,9 +179,11 @@ int s2n_extension_list_parse(struct s2n_stuffer *in, s2n_parsed_extensions_list
     GUARD(s2n_stuffer_init(&extensions_stuffer, &parsed_extension_list->raw));
     GUARD(s2n_stuffer_skip_write(&extensions_stuffer, total_extensions_size));
 
+    uint16_t wire_index = 0;
     while (s2n_stuffer_data_available(&extensions_stuffer)) {
-        GUARD(s2n_extension_parse(&extensions_stuffer, parsed_extension_list->parsed_extensions));
+        GUARD(s2n_extension_parse(&extensions_stuffer, parsed_extension_list->parsed_extensions, &wire_index));
     }
 
+    parsed_extension_list->count = wire_index;
```

Comment:
Could we add a validation check for `wire_index != 0` and wire_index less that the max value for parsed_extensions_list count, prior to assignment here?

Reply:
wire_index == 0 is possible here, if no valid extensions were processed. The max value is the max for uint16_t.